lib/sufia/arkivo/actor.rb

```diff
@@ -0,0 +1,95 @@
+require 'base64'
+require 'tempfile'
+
+# Ugly monkeypatch to make Tempfiles quack like UploadedFiles
+Tempfile.class_eval do
+  attr_accessor :original_filename, :content_type
+end
+
+module Sufia
+  module Arkivo
+    class Actor
+      attr_reader :user, :item
+
+      def initialize(user, item)
+        @user = user
+        @item = item
+      end
+
+      def create_file_from_item
+        batch = Batch.create
+        file_actor = Sufia::GenericFile::Actor.new(::GenericFile.new, user)
+        file_actor.create_metadata(batch.id)
+        store_checksum(file_actor.generic_file)
+        file_actor.create_content(file, item['file']['filename'], file_path, item['file']['contentType'])
+        BatchUpdateJob.new(user.user_key, batch.id, item['metadata']['title'], attributes, default_visibility).run
+        file_actor.generic_file
+      end
+
+      def update_file_from_item(gf)
+        file_actor = Sufia::GenericFile::Actor.new(gf, user)
+        reset_metadata(file_actor)
+        file_actor.update_metadata(attributes, default_visibility)
+        store_checksum(file_actor.generic_file)
+        file_actor.update_content(file, file_path)
+        file_actor.generic_file
+      end
+
+      def destroy_file(gf)
+        gf.destroy
+      end
+
+      private
+
+        def reset_metadata(actor)
+          actor.generic_file.tap do |gf|
```

Comment:
Until we have another use for wiping out metadata *en masse*, I think we should keep this in the Arkivo actor.